### ISSUE TITLE
Updated to R4.0 // getEIA returning NA values

### ISSUE THREAD

**Issue by sawyervillers** [Matt-Brigida/EIAdata#12]
Just updated to R4.0.  Worked fine under 3.6.

Querying a simple data series returns the XTS timevalue but values=NA.
Failure Example:
##################################################
library(EIAdata)
x<-getEIA(ID='NG.NW2_EPG0_SWO_R48_BCF.W',key=privateAPIkey)
##################################################
Can confirm I'm able to query the EIA and return a well formatted JSON object.
Sample code:
##################################################
library(XML)
library(dplyr)

testID<-'NG.NW2_EPG0_SWO_R48_BCF.W'
privateapikey<-yourkeyhere

EIAsingledata <- function(ID,privatekey){

  url <- paste("http://api.eia.gov/series?series_id=",
               ID,
               "&api_key=", privateapikey,
               '&start=20200101',
               "&out=xml",
               sep="" ) 

  doc <- XML::xmlParse(file=url, isURL=TRUE)

  df <- data.frame(stringsAsFactors = F,
                   Date =           as.Date(sapply(doc["//data/row/date"], XML::xmlValue),format='%Y%m%d'),
                   Value =          as.numeric(sapply(doc["//data/row/value"], XML::xmlValue)))

  return(df)
}
######################################################
EIAsingledata(testID,privateapikey)
######################################################
         Date Value
1  2020-05-01  2319
2  2020-04-24  2210
3  2020-04-17  2140
4  2020-04-10  2097
5  2020-04-03  2024
6  2020-03-27  1986
7  2020-03-20  2005
8  2020-03-13  2034
9  2020-03-06  2043
10 2020-02-28  2091
11 2020-02-21  2200
12 2020-02-14  2343
13 2020-02-07  2494
14 2020-01-31  2609
15 2020-01-24  2746
16 2020-01-17  2947
17 2020-01-10  3039
18 2020-01-03  3148

#####VERSION########
platform       x86_64-w64-mingw32          
arch           x86_64                      
os             mingw32                     
system         x86_64, mingw32             
status                                     
major          4                           
minor          0.0                         
year           2020                        
month          04                          
day            24                          
svn rev        78286                       
language       R                           
version.string R version 4.0.0 (2020-04-24)
nickname       Arbor Day  
######SESSION###########
R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17763)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252 LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_0.8.5  XML_3.99-0.3

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6     crayon_1.3.4     assertthat_0.2.1 R6_2.4.1         lifecycle_0.2.0  magrittr_1.5     pillar_1.4.4     rlang_0.4.6      rstudioapi_0.11  vctrs_0.2.4      ellipsis_0.3.0   tools_4.0.0     
[13] glue_1.4.0       purrr_0.3.4      compiler_4.0.0   pkgconfig_2.0.3  tidyselect_1.0.0 tibble_3.0.1

**Comment by Matt-Brigida** [comment]
Thanks for opening the issue.   I'll have to replicate this in a VM, so it may take a day or two.  Does it provide an error or warning message?

**Comment by sawyervillers** [comment]
> Thanks for opening the issue. I'll have to replicate this in a VM, so it may take a day or two. Does it provide an error or warning message?

getEIA() is not throwing an error when it returns NA values.

I stepped through the .getWDEIA function.  NA values are introduced at this line of code:
       `values <- as.numeric(levels(df[,-1]))[df[,-1]]`

Any chance this is related to the fact that R4.0 now enables `stringsAsFactors=F` by default?  
`value = sapply(doc["//data/row/value"], XML::xmlValue)` returns a char type.

I think `levels()` may be expecting a factor instead of a char.  I'm not a programmer, so just theorizing on my part.

**Comment by Matt-Brigida** [comment]
I think you may be correct.  I created a new branch (`issue_12`) and made [this change to the weekly function](https://github.com/Matt-Brigida/EIAdata/commit/f4c9a6e56a984d40e70819a629e1e03da974d12d).  It now should handle both character and factor types.  

If you get a chance it would be great if you can test the fix with R 4.0.  You can install the `issue_12` branch with:

```
library(githubinstall)
gh_install_packages("EIAdata", ref = "issue_12") 
``` 
The fix is now only on the weekly function (so test with some `.W` series).  If it works I'll change the other function and merge to `master`.

**Comment by sawyervillers** [comment]
Matt--Your fix `as.numeric(as.character())` to the weekly function looks good.  Thanks-Dylan

**Comment by Matt-Brigida** [comment]
Great, thanks for testing it.  I applied the fix to the `getEIA()` functions of other frequencies, and merged the `issue_12` branch into `master`. 

This [fix is included in release v0.1.0](https://github.com/Matt-Brigida/EIAdata/releases/tag/v0.1.0). 
 Thanks again for identifying the issue, and have a great day.
